코드리뷰 요청

cba700 opened the pull request:
## 🛰️ Issue Number
코드리뷰 요청

## 🪐 작업 내용


## 📚 Reference


## ✅ Check List
- [ ] 코드가 정상적으로 컴파일되나요?
- [ ] 테스트 코드를 통과했나요?
- [ ] merge할 브랜치의 위치를 확인했나요?
- [ ] Label을 지정했나요?

lunarbae628 committed:
Feat: 회원가입 입력값 유효성 검증 구현

lunarbae628 committed:
Feat: 유효성 검사 예외 처리 수정 (MemberInputInvalidErrorCode 삭제)

zie-ning committed:
Fix: 메세지 보낼 수 있는 길이 늘림

lunarbae628 committed:
Feat: 회원 정보 수정 유효성 검사 구현

cba700 committed:
Refactor: 미사용 컨트롤러 메서드, dto 제거

cba700 committed:
Chore: json 추가

Account-4-Review commented on backend/src/main/java/project/backend/domain/member/api/SignupController.java:
```diff
@@ -0,0 +1,27 @@
+    private final MemberService memberService;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.OK)
```
statusCode 체크 필요

heets-blue commented on backend/src/main/java/project/backend/domain/imagefile/ImageFileService.java:
```diff
@@ -0,0 +1,88 @@
+
+        checkFileExtensionIsImage(extension);
+
+        String storeFileName = UUID.randomUUID() + extension;
```
.을 넣으라는 말씀이신가요? extension은 이미 .을 가지고 있습니다!

lunarbae628 committed:
Fix: 멘토링 코드리뷰 반영 1차

cba700 committed:
Merge branch 'dev' of https://github.com/prgrms-be-devcourse/NBE5-7-2-Team08 into temp

# Conflicts:
#	backend/src/main/java/project/backend/domain/chat/chatroom/api/ChatRoomController.java
#	backend/src/main/java/project/backend/domain/chat/chatroom/app/ChatRoomService.java
#	backend/src/main/java/project/backend/domain/chat/chatroom/mapper/ChatRoomMapper.java
#	backend/src/main/java/project/backend/global/config/security/SecurityConfig.java
#	backend/src/main/java/project/backend/global/exception/GlobalExceptionHandler.java
#	frontend/package-lock.json
#	frontend/src/App.jsx
#	frontend/src/pages/ChatRoom.jsx

cba700 committed:
Fix: 오류 수정

LimKangHyun committed:
refactor: BE에서 넘겨주는 ChatRoom, ChatMessage의 정보를 FE에서 정상적으로 출력가능하도록 변경

zie-ning committed:
Fix: stomp 연결 끊겼을 때 자동 재연결 시도

lunarbae628 committed:
Merge pull request #47 from prgrms-be-devcourse/fix/46-mypage-bug

Fix: 마이페이지 버그 해결

cba700 committed:
Feat: 채팅방 url chat/roomid 에서 chat/roomid/ (복잡한 코드)로 변경했습니다. 이에 따른 프론트도 수정 완료

zie-ning committed:
Fix: stomp 연결 끊겼을 때 재연결 시도 쿨타임 0.5초로 변경

cba700 committed:
Refactor: 컨벤션 적용

cba700 committed:
Feat: 채팅방 나가기 api 기능 구현

LimKangHyun committed:
refactor: SideBar 모달 및 토스트 분리
- 방 생성 및 참가 시, 채팅방 목록 갱신 메서드 추가

cba700 committed:
Feat: 채팅방 나가기 버튼 구현(프론트)

cba700 committed:
Feat: 채팅방 나가면 /Home을 통해 가장 최근 채팅방으로 이동 (프론트)

cba700 committed:
Feat: 채팅방 나가지는 속도 변경

cba700 committed:
Refactor: chatRoomService에서 책임 분리

zie-ning committed:
Fix: stomp 연결 지속 어느정도 완성(이었으면..)

zie-ning committed:
Feat: 특정 채팅방에 참여중인 사용자를 반환하는 api 구현

zie-ning committed:
Fix: stomp 연결이 끊겼을 때 메세지 전송시 alert

lunarbae628 committed:
Merge pull request #49 from prgrms-be-devcourse/feat/48-member-input-valid-check

Feat: 회원가입/정보수정 입력값 유효값 검사 기능

cba700 committed:
Fix: 마이페이지 채팅방 Join 부분 새로운 url로 수정

cba700 committed:
Refactor: DefaultProfileInitializer에 inviteCode도 생성되도록

zie-ning committed:
Fix: 채팅방에 사용자 프로필 일단 디폴트 이미지 뜨게 설정

LimKangHyun committed:
refactor: ChatRoom, SideBar 수정

LimKangHyun committed:
refactor: from에서 of로 메서드명 변경

cba700 committed:
Merge

zie-ning committed:
Fix: 채팅 메세지 앞뒤 공백 유지

zie-ning committed:
Feat: ChatMessageResponse에 senderId 추가

zie-ning committed:
Feat: 채팅방 참여자 프로필 이미지 뜸

zie-ning committed:
Feat: 채팅 이미지 전송 전처리

zie-ning committed:
Refactor: 이미지 저장 경로 변경

zie-ning committed:
Feat: 이미지 저장 경로 수정

zie-ning committed:
Feat: 채팅에서 사진 업로드 가능하게 백엔드 처리

zie-ning committed:
Refactor: 이미지 업로드 고려 프론트 처리

zie-ning committed:
Feat: 이미지 전송 전처리

zie-ning committed:
Feat: 채팅 이미지 전송 완

zie-ning committed:
Design: 이미지 썸네일 구현

zie-ning committed:
Design: 썸네일에서 이미지 삭제 버튼 구현

lunarbae628 committed:
Feat: 깃허브 OAuth인증시 회원가입 성공(불안정함)

lunarbae628 committed:
Fix: JWT토큰 유효기간 수정

zie-ning committed:
Design: 메시지 입력폼 정렬

zie-ning committed:
Feat: 메세지 수정 기능 백엔드 구현

zie-ning committed:
Feat: 이미지 수정 프론트를 구현할 마음의 준비

zie-ning committed:
Feat: 채팅방 입장시 로그인한 유저 정보 가져오기

zie-ning committed:
Feat: 메세지 수정, 삭제 가능한 점 세개 메뉴  추가

zie-ning committed:
Feat: chatMessageResponse에 messageId 추가

zie-ning committed:
Feat: 메세지 수정 ui 추가

zie-ning committed:
Feat: ChatMessageResponse에 edit, delte flag 추가

zie-ning committed:
Feat: 메세지 삭제 백엔드 처리

zie-ning committed:
Feat: 프론트 삭제된 메세지 고려

zie-ning committed:
Feat: 메세지 삭제의 경우 모달창 띄움

zie-ning committed:
Feat: 이미 삭제된 메세지는 메뉴 제공x

zie-ning committed:
Fix: 채팅방 이동시 기존 구독 끊기

zie-ning committed:
Refactor: 찐막

cba700 committed:
Refactor: 코드리뷰 반영

zie-ning committed:
Refactor: 찐찐막

cba700 committed:
Feat: 코드리뷰 반영2

cba700 committed:
Merge pull request #51 from prgrms-be-devcourse/feat/26-join-chatroom-url

Feat/url 보안 강화, url을 통한 입장, 채팅방 나가기 구현

lunarbae628 committed:
Fix: 식별자 깃허브login으로 변경

lunarbae628 committed:
Fix: save for dev merge

lunarbae628 committed:
Fix: 회원가입 오류 수정중

zie-ning committed:
Chore: conflict 해결하기 위한 코드 병합

lunarbae628 committed:
Fix: dev와 merge

lunarbae628 committed:
Feat: JWT 기반 인증 및 재발급 처리 로직 구현

lunarbae628 committed:
Feat: JWT 기반 인증 및 재발급 처리 로직 구현

zie-ning committed:
Refactor: checkMemberExists 함수 제거

zie-ning committed:
Design: searchSidebar 오른쪽옆으로 배치되도록 이동

zie-ning committed:
Merge pull request #55 from prgrms-be-devcourse/hotfix/45-chatting-error-resolve

Fix: 웹소켓 연결 지속 및 채팅 메세지 직렬화

lunarbae628 committed:
Fix: Home.jsx 인증 임시 변경(user/detail)

lunarbae628 committed:
Feat: merge 및 OAuth 가입 유저의 폼 로그인 시도 시 예외 처리 추가

lunarbae628 committed:
Feat: 로그아웃 핸들러 구현

lunarbae628 committed:
Feat: 로그아웃 API 구현

lunarbae628 committed:
Feat: 프론트와 연결

LimKangHyun committed:
Merge branch 'dev' into bugfix/44-chat-error-fix

LimKangHyun committed:
refactor: 채팅방, 채팅방 참여 인원 FE에서 출력

LimKangHyun committed:
fix: ChatRoomController, ChatMessageService 픽스

LimKangHyun committed:
fix: ChatRoomController, ChatMessageService 충돌 해결

fix: ChatRoomController, ChatMessageService 충돌 해결

LimKangHyun committed:
refactor: 채팅방 조회, 채팅방 별 인원 조회 분리

LimKangHyun committed:
refactor: Controller 분리, 응답 DTO에 inviteCode 추가

LimKangHyun committed:
refactor: 채팅방 조회, 채팅방 별 인원 조회 분리

refactor: 채팅방 조회, 채팅방 별 인원 조회 분리

LimKangHyun committed:
feat: 입장 메시지 추가
- messageType에 EVENT 추가

cba700 committed:
Fix: 사이드바 오류 해결 (url관련)

LimKangHyun committed:
fix: EventMessageResponse 수정

lunarbae628 committed:
Fix: 환경변수 설정

LimKangHyun committed:
Feat: 입장 메시지 추가

Feat: 입장 메시지 추가

cba700 committed:
Merge pull request #72 from prgrms-be-devcourse/fix/66-SideBar-fix

Fix: 사이드바 오류 해결 (url관련)

zie-ning committed:
Fix: blankroom url, 프로필 backgroundColor, 깃허브 메세지 ui 수정

lunarbae628 committed:
Fix: 코드리뷰 반영 1

lunarbae628 committed:
Fix: onAuthenticationSuccess에 Transactional추가

lunarbae628 committed:
Merge pull request #65 from prgrms-be-devcourse/feat/52-oauth-jwt

Feat: 깃허브OAuth2 + JWT인증 구현

zie-ning committed:
Merge pull request #74 from prgrms-be-devcourse/fix/61-blankroom-gitmessage

Fix: blankroom url, 프로필 backgroundColor, 깃허브 메세지 ui 수정

LimKangHyun committed:
refactor: 검색 성능 개선
- 검색용 테이블(chat_message_search) 추가
- messageType이 text, code, git 인 경우, chat_message_search에 저장
- chat_message_search테이블에서 room_id의 인덱스 추가(채팅방 검색어 계수 용도, 페이징 용도)
- 입장 메시지 MessageMapper로 분리
- 채팅 수정/삭제 기능 반영

LimKangHyun committed:
refactor: 검색 성능 개선

refactor: 검색 성능 개선

cba700 committed:
Fix: 채팅방 방장은 채팅방 나갈수 없게 수정

cba700 committed:
Fix: 채팅방 나가기 방장은 제외(프론트 수정)

cba700 committed:
Merge pull request #78 from prgrms-be-devcourse/fix/77-owner-not-leave

Fix/77 owner not leave

lunarbae628 committed:
Feat: 깃허브 엑세스 토큰 레디스에 저장

LimKangHyun committed:
fix: full text index 생성 테이블 수정

LimKangHyun committed:
fix: 채팅방 정보 modal 변경

LimKangHyun committed:
Fix: full text index 생성 테이블 수정

Fix: full text index 생성 테이블 수정

cba700 committed:
Feat: 채팅방 입장, 채팅방 생성 시 중간 모달(확인창) 구현

cba700 committed:
Merge branch 'dev' of https://github.com/prgrms-be-devcourse/NBE5-7-2-Team08 into temp

cba700 committed:
Fix: 채팅방 탑바에 채팅방이름 뜨도록 수정

zie-ning committed:
Feat: github 폴더에 있는 클래스 원상복구

lunarbae628 committed:
Fea: 비공개 이메일 가져오기 구현

zie-ning committed:
Init: 불필요한 클래스 제거

zie-ning committed:
Feat: 원상복구가 됐기를

zie-ning committed:
Feat: 서비스 단에서 웹훅 등록

lunarbae628 committed:
Fix: flyway관련 수정

zie-ning committed:
Init: 문성님 pr에 올라온 security 구조로 변경

cba700 committed:
refator: 코드리뷰 반영

cba700 committed:
refactor: 이상한 사진 빼기

cba700 committed:
Merge pull request #82 from prgrms-be-devcourse/feat/81-second-modal

Feat/81 second modal

zie-ning committed:
Chore: yml personal access token 변수 설정 제거

lunarbae628 committed:
Fix: CustomOAuth2User @RequiredArgsConstructor로 변경

lunarbae628 committed:
Merge pull request #83 from prgrms-be-devcourse/feat/75-save-github-accesstoken-get-private-email

Feat: 깃허브 엑세스토큰 레디스 저장과 비공개 이메일 가져오기 구현

zie-ning committed:
Refactor: ChatMessageSearch content 도메인 수정

LimKangHyun committed:
feat: ChatMessageResponse에 isEdited, isDeleted 추가

LimKangHyun committed:
fix: App.css 수정

LimKangHyun committed:
remove: 초대코드 복사버튼 삭제

zie-ning committed:
Fix: merge conflict 해결

LimKangHyun committed:
feat: FE 메시지 수정/삭제 기능 추가

zie-ning committed:
Refactor: github 폴더 통합

zie-ning committed:
Merge pull request #85 from prgrms-be-devcourse/feat/63-webhook-enroll-retry

Feat: 웹훅 자동등록(access token 환경 변수 처리)

zie-ning committed:
Fix 원상복구

zie-ning committed:
Merge pull request #91 from prgrms-be-devcourse/fix/90

Fix 원상복구

LimKangHyun committed:
Merge branch 'dev' into feat/88-message-retain

LimKangHyun committed:
Feat: 메시지 수정/삭제 표시 유지

lunarbae628 committed:
Feat: 토큰 검증 API 구현

lunarbae628 committed:
Feat: 예외처리 일부 수정

lunarbae628 committed:
Save: 중간저장

LimKangHyun committed:
refactor: 채팅 내역 검색 성능 개선

LimKangHyun committed:
refactor: 채팅방 별 채팅 메시지 불러오는 로직 성능 개선

LimKangHyun committed:
fix: 검색 시, 페이지와 페이지에서 보여주는 size 오류 수정

LimKangHyun committed:
fix: MAC netty ERROR 로그 해결
- netty resolver 의존성 주입

lunarbae628 committed:
Feat: 토큰 검증 API 로직 수정

lunarbae628 committed:
Fix: 엑세스토큰 재발급 꼬이는 문제 해결
getAuthentication에서 레디스 토큰검색 생략 validateAuthentication에서만 토큰 레디스 존재여부 판단
추후에 합쳐서 최적화 가능 할 듯

lunarbae628 committed:
Feat: 프론트 인터셉터 구현 완료

LimKangHyun committed:
remove: 중복 의존성 삭제, reactor test 의존성 삭제

LimKangHyun committed:
refactor: reactor test 추가...

LimKangHyun committed:
Refactor: 채팅룸, 채팅내역, 채팅 메시지(검색) 조회 성능 개선

lunarbae628 committed:
MERGE

lunarbae628 committed:
Fix: 사이드바 방생성 예외 처리 수정

lunarbae628 committed:
MERGE

lunarbae628 committed:
Fix: 예외메시지 alert뜨게 수정

lunarbae628 committed:
Fix: 블랭크 참여 / 채팅방 나가기 예외 메시지 alert출력 및ã 채팅방 이름 제대로 출력 찐막

lunarbae628 committed:
Fix: 지은님 코드리뷰 1번 반영

lunarbae628 committed:
Merge pull request #97 from prgrms-be-devcourse/feat/87-axios-interceptors

Feat: interceptors구현 및 토큰 검증 API 구현

cba700 committed:
Fix: 버그 수정, 토근 시간 늘림

cba700 committed:
Merge pull request #99 from prgrms-be-devcourse/fix/bug

Fix: 버그 수정, 토근 시간 늘림

zie-ning committed:
Fix: 깃허브봇 추가

zie-ning committed:
Feat: 토큰 만료 시 로그인으로 리다이렉트되게 시도

zie-ning committed:
Fix: 채팅방 입장 실패시 모달창 사라지게

zie-ning committed:
Fix: 웹훅 등록 실패 시 예외 던져지게 수정

zie-ning committed:
Fix: profile에서 chatroom 입장시 채팅방 이름 뜨게 수정

zie-ning committed:
Merge pull request #101 from prgrms-be-devcourse/fix/94-chatmessage-npe

Fix: 깃허브 메세지 불러들일 때 NPE 나는 오류 해결

zie-ning committed:
Refactor: MessageInput 분리

zie-ning committed:
Refactor: MessageInput 분리 성공!

zie-ning committed:
Refactor: MessageList 분리 시작

LimKangHyun committed:
refactor: TransactionalEventLister 변경

LimKangHyun committed:
refactor: 프로필 이미지 추가

LimKangHyun committed:
refactor: FE 입장메시지, 프로필 이미지, 검색이동 추가

LimKangHyun committed:
feat: FE 채팅방 목록에 알림 추가

LimKangHyun committed:
feat: 채팅방별 인원 프로필 추가

LimKangHyun committed:
feat: FE 채팅방별 인원 프로필 추가

LimKangHyun committed:
Refactor: 입장 이벤트 처리

zie-ning committed:
Refactor: MessageList 분리 완료

zie-ning committed:
Refactor: websocket 연결 부분 분리 완료

zie-ning committed:
Fix: dev와 merge

zie-ning committed:
Refactor: 입장 메세지, 메세지 이동 로직 보완

zie-ning committed:
Fix: useWebSocket useEffect 부분 수정

zie-ning committed:
Refactor: chatRoom 쓸데 없는 코드 컷

zie-ning committed:
Design: 나가기 버튼 roomHeader로 옮김

zie-ning committed:
Refactor: RoomHeader 분리 완료

lunarbae628 committed:
Update README.md

zie-ning committed:
Merge pull request #115 from prgrms-be-devcourse/refactor/102-chatroom

Refactor: 프론트 chatRoom 분리

lunarbae628 committed:
Update README.md

lunarbae628 committed:
Update README.md

lunarbae628 committed:
Update README.md

LimKangHyun committed:
refactor: @ManyToOne 전략 LAZY로 수정

LimKangHyun committed:
refactor: 채팅방이 없는 경우 예외 처리 코드 삭제

LimKangHyun committed:
Refactor: 조인 성능 개선

Refactor: 조인 성능 개선

zie-ning committed:
Refactor: chatmessage가 member를 참조하도록 변경

zie-ning committed:
Refactor: 방장 칼럼을 chatParticipant로 이동

zie-ning committed:
Refactor: 채팅방 이동시마다 memberstatus의 roomId 업데이트

zie-ning committed:
Refactor: 수정, 삭제 flag를 enum 타입으로 변경

lunarbae628 committed:
Refactor: JwtProvider 부분 리펙토링

lunarbae628 committed:
Feat: /auth/refresh 구현 및 리펙토링(테스트 완료)

lunarbae628 committed:
Feat: /auth/Refactor: 인증 예외 관련 리펙토링(커스텀 인증 예외 구현 예정)

lunarbae628 committed:
Feat: /token/sync 실패

lunarbae628 committed:
Fail: /token/sync 작동하나 재발급 자체가 안되는 듯함

zie-ning committed:
Refactor: memberStatus roomId 필드로 변경

lunarbae628 committed:
Refactor: JwtAuthenticationFilter에서 재발급 로직 삭제 및 리펙토링

lunarbae628 committed:
Feat: /token/refresh 구현 및 테스트 완

lunarbae628 committed:
Refactor: 예외 관련 수정

lunarbae628 committed:
Feat: 프론트 인증 인터셉터 구현

cba700 committed:
Refactor: url에서 roomId제거, ChatRoom조회 메서드 수정, 메세지 조회 검증 로직 추가

lunarbae628 committed:
Fix: 오타수정

zie-ning committed:
Refactor: memberStatus 제거 후 member에 recentRoomId 추가

lunarbae628 committed:
Fix: 오타수정 및 토큰 관련 예외 모두 401던지게 수정

lunarbae628 committed:
Fix: @Nonnull 삭제

zie-ning committed:
Refactor: if문 switch문으로 변경

lunarbae628 committed:
Feat: username전환 프론트/회원가입 수정

lunarbae628 committed:
Feat: username 폼 로그인 전환

lunarbae628 committed:
Feat: OAuth2 username(login)으로 전환 성공

lunarbae628 committed:
Feat: email사용 하던 곳 username으로 수정 - 소켓 인증 로직 구현 필요

cba700 committed:
refactor: 코드리뷰 반영

cba700 committed:
Merge pull request #136 from prgrms-be-devcourse/refactor/135-URL-ChatRoom

Refactor: url에서 roomId제거, ChatRoom조회 메서드 수정, 메세지 조회 검증 로직 추가

lunarbae628 committed:
Feat: 소켓 인증 해결

lunarbae628 committed:
Merge pull request #137 from prgrms-be-devcourse/refacotr/132-authentication-full-refactoring

Refactor: JWT인증 리펙토링

zie-ning committed:
Revert "Refactor: if문 switch문으로 변경"

This reverts commit b1bf4b5492ef686295c0321d2817e0c60c0d3221.

sh: line 1: Q: command not found

zie-ning committed:
Merge branch 'dev' into refactor/131-erd-update

zie-ning committed:
Refactor: 방 입장 api와 병합

zie-ning committed:
Fix: 채팅방 참가 안되는 문제 해결

zie-ning committed:
Refactor: editeProfile에서 비밀번호 변경 minLength 4자로 수정

zie-ning committed:
Refactor: 코드리뷰 반영

zie-ning committed:
Merge pull request #134 from prgrms-be-devcourse/refactor/131-erd-update

Refactor: erd 수정사항 반영

LimKangHyun committed:
refactor: 채팅 내역 커서기반 페이징 적용 (무한 스크롤)

LimKangHyun committed:
refactor: 멤버 프로필 정보 변경 시, 채팅 내역에도 적용

LimKangHyun committed:
refactor: roomId 인덱스 삭제
- full text index와 복합으로 불가능하므로 삭제

LimKangHyun committed:
refactor: FE 무한 스크롤, 프로필 업데이트 채팅에 적용

LimKangHyun committed:
refactor: useWebSocket 통합 관리 변경

LimKangHyun committed:
refactor: chatRooms 함수 의존성 배열 제거

LimKangHyun committed:
refactor: 채팅방 인원 갱신 트리거 삭제
- RoomInfoModal 소켓 연결 파트 삭제

LimKangHyun committed:
refactor: 웹 소켓 연결 useWebSocket으로 통합 및 사용자 프로필 및 닉네임 전달안되는 문제 해결

LimKangHyun committed:
refactor: MessageInput 보내주는 부분 onKeyDown 핸들러로 처리 및 ChatRoom에 적용

LimKangHyun committed:
refactor: 자신의 채팅방에 방금 보낸 채팅이 업데이트가 안되는 문제 해결

LimKangHyun committed:
refactor: 컨트롤러 인증 예외 처리 서비스로 이동

LimKangHyun committed:
refactor: 이벤트 객체 정리

LimKangHyun committed:
refactor: 채팅방 참가 확인 메서드 수정

LimKangHyun committed:
refactor: github 웹훅 JWT 필터 패싱 처리

LimKangHyun committed:
refactor: 불필요한 renderWithLink 제거

LimKangHyun committed:
refactor: 스크롤 위치 조정

LimKangHyun committed:
Refactor: 채팅 내역 커서기반 무한스크롤 적용 및 프로필 업데이트 이벤트 처리

Refactor: 채팅 내역 커서기반 무한스크롤 적용

lunarbae628 committed:
Merge: dev와 merge

lunarbae628 committed:
Feat: 웹소켓 연결 단일화 후 소켓 인증 적용)

lunarbae628 committed:
Refactor: 패키지 구조 리펙토링

lunarbae628 committed:
Feat: Username기반에 따라 MyProfile화면 수정

lunarbae628 committed:
Fix: 실수 수정

lunarbae628 committed:
Fix: checkIfMemberExists수정

lunarbae628 committed:
Refactor: 토큰 페이로드 수정 및 토큰 로직 일부 수정

lunarbae628 committed:
Fix: /images/** permitAll 처리

lunarbae628 committed:
Fix: 프론트에 표시되는 Username ID로 변경

lunarbae628 committed:
Fix: email중복 검사 및 프론트 주의메시지 추가

lunarbae628 committed:
Merge pull request #144 from prgrms-be-devcourse/refactor/138-login-with-username

Refactor: Username기반으로 전환 리펙토링
